pkg/nuclide-diagnostics-store/lib/LinterAdapter.js

```diff
@@ -148,35 +148,49 @@ export class LinterAdapter {
   }
 
   async _runLint(editor: TextEditor): Promise<void> {
-    if (this._enabled) {
-      const result = await this._requestSerializer.run(this._provider.lint(editor));
-      if (result.status === 'success') {
-        const buffer = editor.getBuffer();
-        if (buffer.isDestroyed()) {
-          return;
-        }
-
-        if (this._provider.invalidateOnClose && !this._onDestroyDisposables.has(buffer)) {
-          const disposable = buffer.onDidDestroy(() => {
-            this._invalidateBuffer(buffer);
-            this._onDestroyDisposables.delete(buffer);
-            disposable.dispose();
-          });
-          this._onDestroyDisposables.set(buffer, disposable);
-        }
-
-        const linterMessages = result.result;
-        const diagnosticUpdate = linterMessagesToDiagnosticUpdate(
-          editor.getPath(),
-          linterMessages, this._provider.providerName || this._provider.name,
-        );
+    if (!this._enabled) {
+      return;
+    }
+
+    const maybe = this._provider.lint(editor);
+    if (!maybe) {
```

Comment:
Yeah, that's the one exception we make in our config. It's basically due to Flow's `?something` being shorthand for `something | null | undefined`. Since it's used so pervasively we basically have to conflate `null` and `undefined`. Oh well.